motr/ut/idx_dix.c

```diff
@@ -1276,15 +1291,34 @@ static void st_dtm0_r_common(uint32_t sdev_id)
 		return;
 
 	idx_setup();
+	/** PUT redo */
 	exec_one_by_one(1, M0_IC_PUT);
-	dtm0_ut_send_redo(&duc.duc_ifid, sdev_id, &key, &val);
+	dtm0_ut_send_redo(&duc.duc_ifid, sdev_id, &key, &val,
+			  M0_CAS_PUT_FOP_OPCODE);
+
+	/* XXX dirty hack, but now we don't have completion notification */
+	rem = 2ULL * M0_TIME_ONE_SECOND;
+        while (rem != 0)
+                m0_nanosleep(rem, &rem);
+
+	dtm0_ut_read_and_check(key, val, M0_CAS_PUT_FOP_OPCODE);
+
+	idx_teardown();
+
+	/** Delete Redo */
+	idx_setup();
+	exec_one_by_one(1, M0_IC_PUT);
+	exec_one_by_one(1, M0_IC_DEL);
+
+	dtm0_ut_send_redo(&duc.duc_ifid, sdev_id, &key, &val,
```

Comment:
NULL needs to be passed instead of val for consistency of DEL execution, also can be checked in dtm0_ut_cas_op_prepare() to set zero AT buf.